### PR TITLE
added DemandSessionToken and DemandAuthenticatedUser

### PR DESCRIPTION
**Pull request by midahp**
New Middlewares:
- DemandSessionToken: Returns 403 if `Horde-Session-Token` request header is not set or invalid
- DemandAuthenticatedUser: Returns 401 if `HORDE_AUTHENTICATED_USER` request attribute is not set or empty

### CODE REVIEW

File: src/Middleware/DemandAuthenticatedUser.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        $this->responseFactory = $responseFactory;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        if ($request->getAttribute('HORDE_AUTHENTICATED_USER')) {
+            return $handler->handle($request);
+        }
+        return $this->responseFactory->createResponse(401);
```
Comment (ralflang) [inline src/Middleware/DemandAuthenticatedUser.php]: Maybe you want to add a custom message to the 401 response? I'm not sure.

Comment (midahp) [reply src/Middleware/DemandAuthenticatedUser.php]: Makes sense. 'Authenticated user not found.' ?

**Review by ralflang — CHANGES_REQUESTED** [review]
I approve the changes, however we should look into the failing unit tests... 
I think it's a compatibility issue with the horde/log version ^3 installed in the test setup. Maybe it should be horde/log ^3 || dev-FRAMEWORK_6_0 ?

Compare https://github.com/maintaina-com/Http_Server/blob/FRAMEWORK_6_0/.github/workflows/ci.yml

**Comment by ralflang** [comment]
Apart from that, I think start of sentences in comments should be uppercased.
For classes you invented, please also consider adding an author tag

**Pull request merged**